pandas/_libs/lib.pyx

```diff
@@ -1698,6 +1698,20 @@ cpdef bint is_string_array(ndarray values, bint skipna=False):
     return validator.validate(values)
 
 
+cpdef ndarray ensure_string_array(ndarray values, object na_value):
```

Comment:
this should return a new array
and pls
add a doc string

also i think we have a routine like this already